packages/vue-component-meta/src/index.ts

```diff
@@ -99,14 +101,55 @@ export function createComponentMetaChecker(tsconfigPath: string) {
 
 			if ($props) {
 				const type = typeChecker.getTypeOfSymbolAtLocation($props, symbolNode!);
-				const properties = type.getProperties();
-				return properties.map(prop => ({
-					name: prop.escapedName as string,
-					// @ts-ignore
-					isOptional: !!prop.declarations?.[0]?.questionToken,
-					type: typeChecker.typeToString(typeChecker.getTypeOfSymbolAtLocation(prop, symbolNode!)),
-					documentationComment: ts.displayPartsToString(prop.getDocumentationComment(typeChecker)),
-				}));
+				const properties = type.getApparentProperties();
+
+				function reducer (acc: any, cur: any) {
+					acc[cur.name] = cur
+					return acc
+				}
+				function resolveNestedSchema (subtype: ts.Type): PropertyMetaSchema {
+					return (subtype.isClassOrInterface() || subtype.isIntersection())
+						? {
+							kind: 'object',
+							type: typeChecker.typeToString(subtype),
+							schema: subtype.getApparentProperties().map(resolver).reduce(reducer, {})
+						}
+					  : typeChecker.typeToString(subtype)
+				}
+				function resolveArraySchema (subtype: ts.Type): PropertyMetaSchema {
+					// @ts-ignore - typescript dts bug, isArrayType exists
+					return typeChecker.isArrayType(subtype)
+						? {
+							kind: 'array',
+							type: typeChecker.typeToString(subtype),
+							schema: typeChecker.getTypeArguments(subtype as ts.TypeReference).map(resolveSchema)
+						}
+						: resolveNestedSchema(subtype)
+				}
+				function resolveSchema (subtype: ts.Type): PropertyMetaSchema {
+					return subtype.isUnion()
+						? {
+							kind: 'enum',
+							type: typeChecker.typeToString(subtype),
+							schema: subtype.types.map(resolveArraySchema)
+						}
+						: resolveArraySchema(subtype)
+				}
+
+				const resolver = (prop: any): any => {
+					const subtype = typeChecker.getTypeOfSymbolAtLocation(prop, symbolNode!)
+					typeChecker.getDefaultFromTypeParameter(subtype);
+
+					return {
+						name: prop.escapedName,
+						documentationComment: ts.displayPartsToString(prop.getDocumentationComment(typeChecker)),
```

Comment:
@johnsoncodehk @stafyniaksacha I suggest we use standard naming here to facilitate cross usage.

[react-docgen](https://github.com/reactjs/react-docgen) and [vue-docgen-api](https://vue-styleguidist.github.io/docs/Docgen.html) both use the word description here. 

Since we are changing the API foot print anyway, I suggest we make this align with the rest.

```suggestion
						description: ts.displayPartsToString(prop.getDocumentationComment(typeChecker)),
```